backend/src/v1/data/repositories/wordRepository.ts

```diff
@@ -1,36 +1,80 @@
 import { Word } from "../models/models";
-
-interface WordRepository {
-    getWords(filterParams: {}): Word[];
-    getWord(id: string): Word;
-    addWord(data: AddWordData): void;
-    updateWord(word: Word): void;
-    deleteWord(id: string): void;
+import { db } from "../../../db/database";
+import { DeleteResult, InsertResult } from "kysely";
+export interface WordRepository {
+    getWords(languageCode?: string): Promise<Word[]>;
+    getWord(id: number): Promise<Word>;
+    insertWord(data: WordInsert): Promise<{ insertedRows: string; }>;
+    updateWord(data: WordUpdate): Promise<{ updatedRows: string; }>;
+    deleteWord(id: number): Promise<{ deletedRowsCount: string; }>;
 };
 
 const WordRepositoryImpl: WordRepository = {
-    getWords: function (filterParams: {}): Word[] {
-        throw new Error("Function not implemented.");
+    getWords: async (languageCode: string): Promise<Word[]> => {
+        let query = db.selectFrom("word").selectAll();
+        if (languageCode) {
+            query.where("language_code", "=", languageCode);
+        }
+        let result = await query.execute();
+        return result.map((entity) => {
+            return {
+                id: entity.word_id,
+                text: entity.word_text,
+                languageCode: entity.language_code
+            };
+        });
     },
-    getWord: function (id: string): Word {
-        throw new Error("Function not implemented.");
+    getWord: async (id: number): Promise<Word> => {
+        let data = await db.selectFrom("word")
+            .selectAll()
+            .where("word.word_id", "=", id)
+            .executeTakeFirstOrThrow();
+
+        return {
+            id: data?.word_id,
```

Comment:
Watch out for unnecessary optional chaining (`?.`). These can all be `.` instead.

typescript-eslint's [no-unnecessary-condition](https://typescript-eslint.io/rules/no-unnecessary-condition/) rule will catch this sort of thing for you. Here's a recent tweet describing it https://twitter.com/housecor/status/1746550376873025792